tests/st/test_calicoctl.py

```diff
@@ -0,0 +1,1085 @@
+                                         'metadata': {'name': 'profile2'},
+                                         'spec': {
+                                             'egress': [{'action': 'allow',
+                                                         'destination': {},
+                                                         'source': {}}],
+                                             'ingress': [{'ipVersion': 6,
+                                                          'icmp': {'type': 19,
+                                                                   'code': 256},  # max value 255
+                                                          'action': 'deny',
+                                                          'destination': {},
+                                                          'source': {}}],
+                                             'tags': ['tag1', 'tag2s']}}),
+                   ("compound-config", [{
+                       'apiVersion': 'v1',
+                       'kind': 'bgpPeer',
+                       'metadata': {'node': 'Node1',
+                                    'peerIP': '192.168.0.250',
+                                    'scope': 'node'},
+                       'spec': {'asNumber': 64511}},
+                       {'apiVersion': 'v1',
+                        'kind': 'profile',
+                        'metadata': {'name': 'profile2'},
+                        'spec': {
+                            'egress': [{'action': 'allow',
+                                        'destination': {},
+                                        'source': {}}],
+                            'ingress': [{'ipVersion': 6,
+                                         'icmp': {'type': 256,  # 1-byte field
+                                                  'code': 255},
+                                         'action': 'deny',
+                                         'destination': {},
+                                         'source': {}}],
+                            'tags': ['tag1', 'tag2s']},
+                        }],
+                    ),
+               ]
+
+    @parameterized.expand(testdata)
+    def test_invalid_profiles_rejected(self, name, testdata):
+
+        with DockerHost('host', dind=False, start_calico=False) as host:
+            commanderror = False
+            def check_no_data_in_store(testdata):
+                out = host.calicoctl(
+                    "get %s --output=yaml" % testdata['kind'])
+                output = yaml.safe_load(out)
+                assert output == [], "Testdata has left data in datastore " \
+                                     "instead of being completely " \
+                                     "rejected:\n" \
+                                     "Injected: %s\n" \
+                                     "Got back: %s" % (testdata, output)
+
+            host.writefile("testfile.yaml", testdata)
```

Comment:
DockerHost has no attribute writefile.  Should this be self.writefile() instead?